sentry/src/main/java/io/sentry/SentryUUID.java

```diff
@@ -0,0 +1,299 @@
+package io.sentry;
+
+import java.util.Arrays;
+import java.util.Random;
+import java.util.UUID;
+
+/**
+ * Utility class for faster id generation for SentryId, SpanId, and unique filenames uses throughout
+ * the SDK It uses our vendored Random class instead of SecureRandom for improved performance It
+ * directly creates a correctly formatted String to be used as IDs in the Sentry context.
+ *
+ * <p>Id generation is sped up by 4 to 10 times based on the underlying java version.
+ *
+ * <p>Based on the work of <a href="https://github.com/jchambers/">Jon Chambers</a> Here:
+ * https://github.com/jchambers/fast-uuid
+ */
+public class SentryUUID {
+
+  private static final int UUID_STRING_LENGTH = 36;
+  private static final int SENTRY_UUID_STRING_LENGTH = 32;
+  private static final int SENTRY_SPAN_UUID_STRING_LENGTH = 16;
+
+  private static final char[] HEX_DIGITS =
+      new char[] {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'};
+
+  private static final long[] HEX_VALUES = new long[128];
+
+  static {
+    Arrays.fill(HEX_VALUES, -1);
+
+    HEX_VALUES['0'] = 0x0;
+    HEX_VALUES['1'] = 0x1;
+    HEX_VALUES['2'] = 0x2;
+    HEX_VALUES['3'] = 0x3;
+    HEX_VALUES['4'] = 0x4;
+    HEX_VALUES['5'] = 0x5;
+    HEX_VALUES['6'] = 0x6;
+    HEX_VALUES['7'] = 0x7;
+    HEX_VALUES['8'] = 0x8;
+    HEX_VALUES['9'] = 0x9;
+
+    HEX_VALUES['a'] = 0xa;
+    HEX_VALUES['b'] = 0xb;
+    HEX_VALUES['c'] = 0xc;
+    HEX_VALUES['d'] = 0xd;
+    HEX_VALUES['e'] = 0xe;
+    HEX_VALUES['f'] = 0xf;
+
+    HEX_VALUES['A'] = 0xa;
+    HEX_VALUES['B'] = 0xb;
+    HEX_VALUES['C'] = 0xc;
+    HEX_VALUES['D'] = 0xd;
+    HEX_VALUES['E'] = 0xe;
+    HEX_VALUES['F'] = 0xf;
+  }
+
+  private SentryUUID() {
+    // A private constructor prevents callers from accidentally instantiating FastUUID objects
+  }
+
+  /**
+   * Parses a UUID from the given character sequence. The character sequence must represent a UUID
+   * as described in {@link UUID#toString()}.
+   *
+   * @param uuidSequence the character sequence from which to parse a UUID
+   * @return the UUID represented by the given character sequence
+   * @throws IllegalArgumentException if the given character sequence does not conform to the string
+   *     representation as described in {@link UUID#toString()}
+   */
+  public static UUID parseUUID(final CharSequence uuidSequence) {
```

Comment:
Not sure if we even need this

Reply:
I presume no, as long as we use String values in the SDK.